tests/acceptance/bootstrap/NotificationContext.php

```diff
@@ -513,20 +513,40 @@ public function userShouldHaveReceivedTheFollowingEmailFromUser(string $user, st
 		$this->assertEmailContains($user, $expectedEmailBodyContent);
 	}
 
+	/**
+	 * @Then user :user should have received the following email from user :sender ignoring extra spaces and blank lines
+	 *
+	 * @param string $user
+	 * @param string $sender
+	 * @param PyStringNode $content
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function userShouldHaveReceivedTheFollowingEmailFromUserIgnoringExtraSpaces(string $user, string $sender, PyStringNode $content):void {
+		$rawExpectedEmailBodyContent = \str_replace("\r\n", "\n", $content->getRaw());
+		$expectedEmailBodyContent = $this->featureContext->substituteInLineCodes(
+			$rawExpectedEmailBodyContent,
+			$sender
+		);
+		$this->assertEmailContains($user, $expectedEmailBodyContent, true);
+	}
+
 	/***
 	 * @param string $user
 	 * @param string $expectedEmailBodyContent
+	 * @param bool $ignoreExtraSpacesAndBlankLines
 	 *
 	 * @return void
 	 * @throws GuzzleException
 	 */
-	public function assertEmailContains(string $user, string $expectedEmailBodyContent):void {
+	public function assertEmailContains(string $user, string $expectedEmailBodyContent, $ignoreExtraSpacesAndBlankLines = false):void {
 		$address = $this->featureContext->getEmailAddressForUser($user);
 		$this->featureContext->pushEmailRecipientAsMailBox($address);
 		$actualEmailBodyContent = EmailHelper::getBodyOfLastEmail($address, $this->featureContext->getStepLineRef());
 		Assert::assertStringContainsString(
```

Comment:
move it to a if block. improves the code readability
```suggestion
		$actualEmailBodyContent = EmailHelper::getBodyOfLastEmail($address, $this->featureContext->getStepLineRef());

		if (ignoreWhiteSpace) {
			$expectedEmailBodyContent = preg_replace('/(^\s+|\s+$)/m', '', $expectedEmailBodyContent)
			$actualEmailBodyContent = preg_replace('/(^\s+|\s+$)/m', '', $actualEmailBodyContent)
		}

		Assert::assertStringContainsString(
```